src/cryptonote_core/blockchain.cpp

```diff
@@ -351,6 +351,11 @@ bool Blockchain::init(BlockchainDB* db, const network_type nettype, bool offline
 
   m_nettype = test_options != NULL ? FAKECHAIN : nettype;
   m_offline = offline;
+  m_regtest = static_cast<bool>(regtest_diff);
+  if (m_regtest)
+  {
+    m_fixed_difficulty = *regtest_diff;
+  }
```

Comment:
That seems unfortunate. I often hack a "return constant" for difficulty on testnet. It's a useful thing to have, and this new change limits it to regtest mode.

Reply:
OK, I made this patch to accommodate your request: https://github.com/stoffu/monero/commit/1f283ac93a12b8dc910cdc270c0ba15fe4471126